cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -217,11 +217,10 @@ protected AbstractContext(Program program, ProgramOptions programOptions, CConfi
 
   private MetricsCollectionService getMetricsService(CConfiguration cConf, MetricsCollectionService metricsService,
                                                      Map<String, String> runtimeArgs) {
-    String emitMetricsPreference = runtimeArgs.get(Constants.Metrics.EMIT_PROGRAM_CONTAINER_METRICS);
-
-    // Get the option for emitting metrics from preferences or as a fall back from cConf
-    boolean emitMetrics = emitMetricsPreference != null ? Boolean.valueOf(emitMetricsPreference) :
-      cConf.getBoolean(Constants.Metrics.EMIT_PROGRAM_CONTAINER_METRICS);
+    boolean emitMetrics = cConf.getBoolean(Constants.Metrics.EMIT_PROGRAM_CONTAINER_METRICS);
+    if (runtimeArgs.containsKey(SystemArguments.EMIT_PROGRAM_CONTAINER_METRICS_PREFERENCE)) {
+      emitMetrics = SystemArguments.isProgramMetricsEnabled(runtimeArgs);
```

Comment:
Better have the `isProgramMetricsEnabled` takes a default value, the code would be much cleaner.

```java
boolean emitMetrics = SystemArguments.isMetricsEnabled(runtimeArgs, cConf.getBoolean(Constants.Metrics.METRICS_ENABLED));
```

Reply:
👍